rest-api/config_api.py

```diff
@@ -64,45 +63,60 @@ def check_config_admin():
   raise Forbidden()
 
 
-class ConfigApi(base_api.BaseApi):
+class ConfigApi(Resource):
   """Api handlers for retrieving and setting config values."""
-
   method_decorators = [auth_required_config_admin]
 
-  def __init__(self):
-    super(ConfigApi, self).__init__(config.ConfigurationDAO())
-
-  def get_config_by_date(self, key, date):
-    result = config.get_config_that_was_active_at(key, api_util.parse_date(date))
-    return self.make_response_for_resource(self.dao.to_json(result))
-
-  def get(self, key=None):
+  def get(self, key=config.CONFIG_SINGLETON_KEY):
     date = request.args.get('date')
-    key = key or config.CONFIG_SINGLETON_KEY
-    if not date:
-      # Return the live config.
-      return super(ConfigApi, self).get(key)
-    else:
-      return self.get_config_by_date(key, date)
-
-  # Insert or update to configuration
-  def post(self, key=None):
-    key = key or config.CONFIG_SINGLETON_KEY
-    resource = request.get_json(force=True)
-    m = self.dao.from_json(resource, None, key)
-    self.validate_object(m, None)
-    self.dao.store(m)
-    return self.make_response_for_resource(self.dao.to_json(m))
+    if date is not None:
+      date = parse_date(date)
+    model = config.load(key, date=date)
+    data = model.configuration or {}
+    return self.make_response_for_resource(data)
 
-  def put(self, key=None):
-    return super(ConfigApi, self).put(key or config.CONFIG_SINGLETON_KEY)
-
-  def validate_object(self, obj, a_id=None):
-    super(ConfigApi, self).validate_object(obj, a_id)
-    if obj.key.id() != config.CONFIG_SINGLETON_KEY:
+  def post(self, key=config.CONFIG_SINGLETON_KEY):
```

Comment:
Seems like post and put should share the same implementation,differing only by the lookup for existing config?

Reply:
not quite - `put` also does some stuff with `date` and `client_id`.  They are similar though; I've managed to reduce them a little more.